buildbot/configure.py

```diff
@@ -322,6 +326,11 @@ def main():
     parser.add_argument(
         "--shared-libs", action="store_true", help="Build shared libraries"
     )
+    parser.add_argument(
+        "--gnu-toolchain",
+        metavar="GNU_TOOLCHAIN",
+        help="Toolchain for compiler driver result",
```

Comment:
This message might be confusing. This argument passed via `-DSYCL_LIBDEVICE_GCC_TOOLCHAIN`, which (I think) configures GCC toolchain only for `libdevice` project. All other projects will use clang's default configuration. Right?

Reply:
Theoretically, the `SYCL_LIBDEVICE_GCC_TOOLCHAIN` should indeed only apply to the `libdevice`.

However practically, this is the only way I was able get the follow-on `compile.py` to complete. Without this option, the `clang` executables are created correctly. But then those executables fail when deployed for the remaining part of the build: The `clang` executables keep grabbing the system library for `libstdc++` (which fails).

With this setting, we're getting our `clang` compiler to correctly pick up the desired `libstdc++` library...

Reply:
AFAIK, this option doesn't impact the way `clang` executables are created, but how they are used in `libdevice` project. This project uses clang to build SYCL libraries. But help message for `--gnu-toolchain` argument might be interpreted as it changes the location of the GNU toolchain clang picks up by default. I think we should clarify that this option is intended to impact GNU toolchain location for `libdevice` project only.

Reply:
Will do;
I am just getting up to speed on this project, where can I get more details on the different components that we're building here?

Reply:
> where can I get more details on the different components that we're building here?

I don't think we have a separate document answering this question, so the source code for the `configure.py` and CMake files is the best source of information.

I also recommend taking a look at project's documentation - https://intel.github.io/llvm-docs/.